[WIP] Improve Delegation Screen

aliseyalvi opened the pull request:
### What does this PR?

- Added autocomplete dropdown for users when typing in ' to ' input field.
- Added input for selecting amount to be sent and synced with the slider.

Note: Work is in progress. Needs further work and testing.

### Where should the reviewer start?

### Steps to reproduce
Go to Wallet page.
swipe to last of item in horizontal list named Hive Power.
Tap on dropdown arrow and open delegation screen.

### Issue number
#2158 

### Screenshots/Video

https://www.loom.com/share/99a56a35030a43e7ad9fcc29572c6929

noumantahir commented:
@aliseyalvi there is a UserAvatar component that render user's profile picture if you give it username. we should show that as well along with user names returned by api

aliseyalvi commented:
@noumantahir check the latest update from here https://github.com/ecency/ecency-mobile/pull/2179/commits/6400d5075305b017e6ae31aeae6f545ac83ed617

https://www.loom.com/share/03bac1c0699045608b45c712042f63ba

noumantahir commented:
@aliseyalvi functionality wise seems to be going in right direction. only thing I see missing is fetching the previous hive power delegated to same user

On the UI side, a lot can be improved, will see what ideas I can come up with

noumantahir reviewed (commented):
show previously delegated HP, even if it's 0... show simple text for now... will think UI next

<img width="345" alt="Screenshot 2022-02-08 at 1 01 28 AM" src="https://user-images.githubusercontent.com/6298342/152862749-16dbae2f-1407-4b85-8535-4f2ef2a9e735.png">

aliseyalvi commented:
@noumantahir added already delegated and remaining hp details on delegation screen.
Also added amount validation check and disable the next step if amount entered is 0 or more than available hp.
Please review.

aliseyalvi commented:
@noumantahir please check the latest update.
Updated according to new design. 
Check the screen recording below.

https://user-images.githubusercontent.com/48380998/153700415-a2398f83-b848-4803-9b23-4acf5ad760c6.mp4

noumantahir commented:
@aliseyalvi great going overall... here are a few comments with screenshots....Let me know once done, I will be doing some final touchups on margins and text spacing afterwards.

<img width="350" alt="Screenshot 2022-02-12 at 11 59 48 PM" src="https://user-images.githubusercontent.com/6298342/153725061-fa283efa-47a5-4b1a-9af5-a237de8ec658.png">
<img width="354" alt="Screenshot 2022-02-13 at 12 02 02 AM" src="https://user-images.githubusercontent.com/6298342/153725066-7e4b3388-d8bd-4edd-9450-74ee87e56f64.png">
<img width="351" alt="Screenshot 2022-02-13 at 12 03 38 AM" src="https://user-images.githubusercontent.com/6298342/153725069-ec09b38c-c859-4456-a06e-1119e674c500.png">
<img width="351" alt="Screenshot 2022-02-13 at 12 07 06 AM" src="https://user-images.githubusercontent.com/6298342/153725072-f2c97665-a728-43b7-b0cf-10c2b7ee78d7.png">

aliseyalvi commented on src/redux/actions/uiAction.ts:
```diff
@@ -23,13 +23,15 @@ export const toastNotification = (payload:string) => ({
 });
 
-export const showActionModal = (title:string, body?:string, buttons?:AlertButton[], headerImage?:any, onClosed?:()=>void) => ({
+export const showActionModal = (title:string, body?:string, para?: string, buttons?:AlertButton[], headerContent?:any, headerImage?:any, onClosed?:()=>void) => ({
```
@noumantahir changed actionModal implementation and converted to options object. https://github.com/ecency/ecency-mobile/pull/2179/commits/3aa9fec9a61fed8118e1d5e5f2351cc3f5e02752

aliseyalvi replied:
tested other action modal implementation. working well. only didn't tested the app update modal .

aliseyalvi commented on src/screens/transfer/screen/delegateScreen.js:
```diff
@@ -101,6 +171,63 @@ class DelegateScreen extends Component {
+      dispatch(
+        showActionModal(
+          intl.formatMessage({ id: 'transfer.confirm' }),
+          intl.formatMessage(
```
actually we have to show them conditionally and they have to pass through redux so  it would be very complex. para props seems a straigh forward approach and less complex. Also we can show it conditionally

aliseyalvi commented:
> fixed already delegated amount not changing bug

fixed this bug here https://github.com/ecency/ecency-mobile/pull/2179/commits/3442e12e241a61e4afeaf1681c8db060844929d1.
It was due to method calling in life cycle method conditionally. Removed the method from life cycle and called it in user name selection. It will fetch each time when new user will be selected.

aliseyalvi commented:
@noumantahir fixed the loophole of selecting same users in from and to inputs https://github.com/ecency/ecency-mobile/pull/2179/commits/2d43a4df3d496ef1028db0bb40df6b32dbe0a99a.
Also fixed a bug causing not changing already delegated amount when from input changes. https://github.com/ecency/ecency-mobile/pull/2179/commits/d5bdf24905eb1ed8691185982b12d2b956bf5f32

aliseyalvi commented:
@noumantahir disabled user account switch in delegation process. didn't removed the dropdown for later implementation https://github.com/ecency/ecency-mobile/pull/2179/commits/418d9a630bd191da97c91d90a224369ac2f05d5a

noumantahir commented:
@aliseyalvi there was a consistent bug that I just fixed, do checkout the change I made to avoid it in future... to summarise, setState or useState's set methods are always asynchronous their value is updated with a slight delay and we have to handle this behaviour accordingly...

![Screenshot 2022-02-16 at 11 33 51 AM](https://user-images.githubusercontent.com/6298342/154209778-b2f93e3d-e203-4c07-9ca2-71d39a6b7f13.png)

noumantahir commented:
@feruzm you may review and merge if you feel it matches Ecency standards...
